pandas/tests/io/test_iceberg.py

```diff
@@ -0,0 +1,143 @@
+"""
+Tests for the Apache Iceberg format.
+
+Tests in this file use a simple Iceberg catalog based on SQLite, with the same
+data used for Parquet tests (``pandas/tests/io/data/parquet/simple.parquet``).
+"""
+
+import collections
+import importlib
+import pathlib
+
+import pytest
+
+import pandas as pd
+import pandas._testing as tm
+
+from pandas.io.iceberg import read_iceberg
+
+pytestmark = [pytest.mark.single_cpu]
+
+pyiceberg = pytest.importorskip("pyiceberg")
+pyiceberg_catalog = pytest.importorskip("pyiceberg.catalog")
+pq = pytest.importorskip("pyarrow.parquet")
+
+Catalog = collections.namedtuple("Catalog", ["name", "uri"])
+
+
+@pytest.fixture
+def catalog(request, tmp_path):
+    # the catalog stores the full path of data files, so the catalog needs to be
+    # created dynamically, and not saved in pandas/tests/io/data as other formats
+    uri = f"sqlite:///{tmp_path}/catalog.sqlite"
+    warehouse = f"file://{tmp_path}"
+    catalog_name = request.param if hasattr(request, "param") else None
+    catalog = pyiceberg_catalog.load_catalog(
+        catalog_name or "default",
+        type="sql",
+        uri=uri,
+        warehouse=warehouse,
+    )
+    catalog.create_namespace("ns")
+
+    df = pq.read_table(
+        pathlib.Path(__file__).parent / "data" / "parquet" / "simple.parquet"
+    )
+    table = catalog.create_table("ns.my_table", schema=df.schema)
+    table.append(df)
+
+    if catalog_name is not None:
+        config_path = pathlib.Path.home() / ".pyiceberg.yaml"
```

Comment:
Can this be written in the `tmp_path` too? Otherwise, I imagine after running the tests this will be leftover

Reply:
This file is being removed in line 64: https://github.com/pandas-dev/pandas/pull/61383/files#diff-dc72941f90793049e0faeefb11e9e743bd46697b5d8b44be927c8c3961609914R64

To be able to write it in `tmp_path` I need to set up an environment variable for pyiceberg to find it. I thought this was easier, but I can do that if you have a strong preference for it.

Reply:
Ah sorry I missed this. What you have now is fine